zebra/zebra_dplane.c

```diff
@@ -7420,16 +7449,54 @@ static void dplane_thread_loop(struct event *event)
 		if (!zdplane_info.dg_run)
 			break;
 
+		/* Locate next provider */
+		next_prov = dplane_prov_list_next(&zdplane_info.dg_providers,
+						  prov);
+		if (next_prov) {
+			curr = atomic_load_explicit(&next_prov->dp_in_queued,
+						    memory_order_relaxed);
+			out_curr =
+				atomic_load_explicit(&next_prov->dp_out_queued,
+						     memory_order_relaxed);
+		} else
+			out_curr = curr = 0;
+
 		/* Dequeue completed work from the provider */
 		dplane_provider_lock(prov);
 
-		while (counter < limit) {
-			ctx = dplane_provider_dequeue_out_ctx(prov);
-			if (ctx) {
-				dplane_ctx_list_add_tail(&work_list, ctx);
-				counter++;
-			} else
-				break;
+		if (curr >= (uint64_t)limit) {
+			reschedule = true;
+			if (IS_ZEBRA_DEBUG_DPLANE_DETAIL)
+				zlog_debug("%s: Next Provider(%s) Input queue is %" PRIu64
```

Comment:
IMO, printing `next_prov->dp_name` would be useful here, otherwise hard to distinguish?

Reply:
we are printing it?